api/src/main/java/com/datastrato/gravitino/authorization/SecurableObjects.java

```diff
@@ -188,26 +263,60 @@ public boolean equals(Object other) {
 
       SecurableObject otherSecurableObject = (SecurableObject) other;
       return Objects.equals(parent, otherSecurableObject.parent())
-          && Objects.equals(name, otherSecurableObject.name());
+          && Objects.equals(name, otherSecurableObject.name())
+          && Objects.equals(type, otherSecurableObject.type());
     }
   }
 
   /**
    * Create a {@link SecurableObject} from the given identifier string.
    *
    * @param securableObjectIdentifier The identifier string
+   * @param type The securable object type.
    * @return The created {@link SecurableObject}
    */
-  public static SecurableObject parse(String securableObjectIdentifier) {
+  public static SecurableObject parse(String securableObjectIdentifier, SecurableObject.Type type) {
```

Comment:
It's a bit unknown about this `securableObjectIdentifier`, so I would suggest not introducing new name/concept, you can use "fullName" and add more comments.